commons/src/main/java/org/infinispan/commons/marshall/WrappedByteArray.java

```diff
@@ -63,25 +56,19 @@ public boolean equals(Object o) {
    }
 
    public boolean equalsWrappedBytes(WrappedBytes other) {
-      int length = getLength();
-      if (other.getLength() != length) return false;
-      if (other.hashCode() != hashCode()) return false;
-      for (int i = 0; i < length; ++i) {
-         if (getByte(i) != other.getByte(i)) return false;
-      }
-      return true;
+      //Array.equals already checks the length and hashcode is lazy now
+      return Arrays.equals(bytes, other.getBytes());
```

Comment:
Actually I take that back, this shouldn't be changed.  The purpose of this method is to allow for something without a backing array to test equality.  This will be useful when off heap comes back with an instance.  I would say either leave it alone, throw an exception or remove this method.